foxpuppet/windows/browser/tab.py

```diff
@@ -0,0 +1,39 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.region import Region
+
+
+class Tab(Region):
+
+    _tabs_locator = (By.TAG_NAME, 'tab')
+
+    @property
+    def close_button(self):
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tab = self.selenium.find_element(*self._tabs_locator)
+            return tab.find_anonymous_element_by_attribute(
+                'anonid', 'close-button')
+
+    @property
+    def switch_to(self):
+        self.selenium.switch_to.window(self.selenium.current_window_handle)
+
+    def close(self):
+        from foxpuppet.windows.browser.tab_bar import TabBar
+        from foxpuppet.windows.browser.window import BrowserWindow
+
+        window = BrowserWindow(
+            self.selenium, self.selenium.current_window_handle)
+
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tabbar = TabBar(window, Tab)
+            current_tabs = tabbar.tabs
```

Comment:
I see. So you don't want to fallback to `window_handles` but do really everything via the UI elements? In such a case we might indeed need a weakref to the `tabtab` class.